experimental/LieAlgebras/src/LieAlgebraIdeal.jl

```diff
@@ -0,0 +1,283 @@
+  ) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+    return LieAlgebraIdeal{C,LieT}(L, Vector{LieT}(map(L, gens)); kwargs...)
+  end
+end
+
+###############################################################################
+#
+#   Basic manipulation
+#
+###############################################################################
+
+base_lie_algebra(
+  I::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}} = I.base_lie_algebra::parent_type(LieT)
+
+function gens(I::LieAlgebraIdeal)
+  return I.gens
+end
+
+function gen(I::LieAlgebraIdeal, i::Int)
+  return I.gens[i]
+end
+
+function ngens(I::LieAlgebraIdeal)
+  return length(gens(I))
+end
+
+function basis_matrix(
+  I::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  return I.basis_matrix::dense_matrix_type(C)
+end
+
+@doc raw"""
+    basis(I::LieAlgebraIdeal) -> Vector{LieAlgebraElem}
+
+Return the basis of the ideal `I`.
+"""
+function basis(I::LieAlgebraIdeal)
+  return I.basis_elems
+end
+
+@doc raw"""
+    basis(I::LieAlgebraIdeal, i::Int) -> LieAlgebraElem
+
+Return the `i`-th basis element of the ideal `I`.
+"""
+function basis(I::LieAlgebraIdeal, i::Int)
+  return I.basis_elems[i]
+end
+
+@doc raw"""
+    dim(I::LieAlgebraIdeal) -> Int
+
+Return the dimension of the ideal `I`.
+"""
+dim(I::LieAlgebraIdeal) = length(basis(I))
+
+###############################################################################
+#
+#   String I/O
+#
+###############################################################################
+
+function Base.show(io::IO, ::MIME"text/plain", I::LieAlgebraIdeal)
+  io = pretty(io)
+  println(io, LowercaseOff(), "Lie algebra ideal")
+  println(io, Indent(), "of dimension $(dim(I))", Dedent())
+  if dim(I) != ngens(I)
+    println(io, Indent(), "with $(ItemQuantity(ngens(I), "generator"))", Dedent())
+  end
+  print(io, "over ")
+  print(io, Lowercase(), base_lie_algebra(I))
+end
+
+function Base.show(io::IO, I::LieAlgebraIdeal)
+  io = pretty(io)
+  if get(io, :supercompact, false)
+    print(io, LowercaseOff(), "Lie algebra ideal")
+  else
+    print(io, LowercaseOff(), "Lie algebra ideal over ", Lowercase())
+    print(IOContext(io, :supercompact => true), base_lie_algebra(I))
+  end
+end
+
+###############################################################################
+#
+#   Comparisons
+#
+###############################################################################
+
+function Base.issubset(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  @req base_lie_algebra(I1) === base_lie_algebra(I2) "Incompatible Lie algebras."
+  return all(in(I2), gens(I1))
+end
+
+function Base.:(==)(
+  I1::LieAlgebraIdeal{C,LieT}, I2::LieAlgebraIdeal{C,LieT}
+) where {C<:RingElement,LieT<:LieAlgebraElem{C}}
+  base_lie_algebra(I1) === base_lie_algebra(I2) || return false
+  gens(I1) == gens(I2) && return true
+  return basis_matrix(I1) == basis_matrix(I2)
```

Comment:
(See the remark about equality of subalgebras.)

Reply:
see above. same change applied as for ideals